pkg/provider/kubernetes/auth_test.go

```diff
@@ -250,28 +250,13 @@ func TestSetAuth(t *testing.T) {
 						"cert": []byte("my-cert"),
 						"key":  []byte("my-key"),
 					},
-				}, &corev1.Secret{
```

Comment:
do not delete these

Reply:
this I have removed because than it reflects exactly why we have this bug in. Removing the token here is like not providing a token just a key/cert. Do you want me to create an additional step rather adjusting this one?

Reply:
>this I have removed because than it reflects exactly why we have this bug in.

So yes, keep it in and expect it to fail.

Reply:
We have to clarify the difference between the original implementation and how we want to have it in future.

The original one did take the token from a secret or from a given `ServiceAccount` and in addition it appended the key/cert when given.

The new approach would take just one, and the prio would be, 1. token from secret, 2. token from `ServiceAccount` and 3. the last one the key/cert pair.

I do wonder if somebody is interested in having a token and a key/cert in the config together. Feels weird but I do not know what the kube client does if it has multiple options to authenticate with.

The question is now if we should stick to the original implementation or not.